### ISSUE TITLE
setPosition does not work with fonts

### ISSUE THREAD

**Issue by dogunbound** [SFML/SFML#2172]
Thanks for raising your issue here! 🙂

Before you submit the issue however, we'd like you to consider the follow points.

* We like to use the issue tracker for confirmed issues.
* If you're stuck with SFML, please use [the forum](https://en.sfml-dev.org/forums/index.php#c3) to get help.

----

## Subject of the issue

setPosition applies an erroneous behind the scene offset when it is ran.

## Your environment

* linux / x11
* SFML 2.5.1
* g++ main.cpp -o sfml-app -lsfml-graphics -lsfml-window -lsfml-system && ./sfml-app 

## Steps to reproduce

Tell us how to reproduce this issue. Please provide a [minimal, complete and verifiable example](https://stackoverflow.com/help/mcve), you can use the follow template as a start:

```cpp
#include <SFML/Graphics.hpp>
#include <SFML/System.hpp>
#include <iostream>

int main() {
  sf::Vector2f pos(25, 25);

  sf::RenderWindow window(sf::VideoMode(800, 600), "SFML works!");

  sf::Font font;
  font.loadFromFile("arial 1.ttf");

  sf::Text text;
  text.setFont(font);
  text.setString("Hello World!");
  text.setCharacterSize(48);
  text.setFillColor(sf::Color::White);
  // set pos done here
  text.setPosition(pos); 
  sf::RectangleShape shape(sf::Vector2f(text.getGlobalBounds().width, text.getGlobalBounds().height));
  // and here
  shape.setPosition(pos);
  shape.setFillColor(sf::Color::Transparent);
  shape.setOutlineColor(sf::Color::Magenta);
  shape.setOutlineThickness(2);

  // These should all be the same!
  std::cout << text.getGlobalBounds().top << " " 
    << text.getGlobalBounds().left << " "
    << shape.getGlobalBounds().top << " " 
    << shape.getGlobalBounds().left << " " 
    << std::endl;

  while (window.isOpen())
  {
    sf::Event event;
    while (window.pollEvent(event))
    {
      if (event.type == sf::Event::Closed)
        window.close();
    }

    window.clear();
    window.draw(text);
    window.draw(shape);
    window.display();
  }

  return 0;
}

```

## Expected behavior

Tell us what should happen

## Actual behavior

Tell us what happens instead

**Comment by ChrisThrasher** [comment]
Can you share the font .tff you're using so we can be sure that isn't the cause of any of these issues?

**Comment by dogunbound** [comment]
[Archive.zip](https://github.com/SFML/SFML/files/9214318/Archive.zip)

Both fonts exhibit similar behavior @ChrisThrasher

**Comment by Bambo-Borris** [comment]
I don't think this is a bug, I think this is intended. The global bounds of a `sf::Text` is the result of its local bounds being transformed and the default local bounds don't have a {left, top} of {0,0}. But a RectangleShape does have a default {left, top} of {0,0}

**Comment by dogunbound** [comment]
After talking to @Bambo-Borris on discord. He is probably right. The real issue is there is NO documentation on having to deal with the font's inbuilt buffer. That's why I thought this was a bug.

I would much prefer a solution where you can disable this offset, but idk how feasible this is.

**Comment by dogunbound** [comment]
After some more tinkering, I managed to find a solution to the problem that undoes the text buffer.

```rust
pub fn create_unbuffered_text<'a>(
    resource_manager: &'a ResourceManager,
    string: &String,
    font_size: u32,
) -> Text<'a> {
    let mut text = Text::new(string, &resource_manager.fetch_current_font(), font_size);
    let local_bounds = text.local_bounds().position();
    text.set_origin(local_bounds);
    text
}
```


There could be a very simple "resetOrigin" function, or something along those lines that does exactly this. Or another constructor that undoes the buffer.

**Comment by dogunbound** [comment]
One more thing to note, I just find it odd to do:
```c++
sf::Vector2f pos(25, 25);
sf::Font font;
font.loadFromFile("arial 1.ttf");

sf::Text text;
text.setFont(font);
text.setString("Hello World!");
text.setPosition(pos); 

std::cout << text.getGlobalBounds().top << " " 
  << text.getGlobalBounds().left << " "
  << shape.getGlobalBounds().top << " " 
  << shape.getGlobalBounds().left << " " 
  << std::endl;
```

And the global positions not to be the same. It just feels wrong.

**Comment by Bambo-Borris** [comment]
> One more thing to note, I just find it odd to do:
> 
> ```c++
> sf::Vector2f pos(25, 25);
> sf::Font font;
> font.loadFromFile("arial 1.ttf");
> 
> sf::Text text;
> text.setFont(font);
> text.setString("Hello World!");
> text.setPosition(pos); 
> 
> std::cout << text.getGlobalBounds().top << " " 
>   << text.getGlobalBounds().left << " "
>   << pos.y << " " 
>   << pos.x << " " 
>   << std::endl;
> ```
> 
> And the global positions not to be the same. It just feels wrong.

But as explained, the position you're setting will relate to the origin. And the global bounds left & top position aren't the origin. Just like if you centre the origin on a square, then use `getGlobalBounds()` the left & top positions will not refer to the origin position of the shape.

**Comment by dogunbound** [comment]
> > One more thing to note, I just find it odd to do:
> > ```c++
> > sf::Vector2f pos(25, 25);
> > sf::Font font;
> > font.loadFromFile("arial 1.ttf");
> > 
> > sf::Text text;
> > text.setFont(font);
> > text.setString("Hello World!");
> > text.setPosition(pos); 
> > 
> > std::cout << text.getGlobalBounds().top << " " 
> >   << text.getGlobalBounds().left << " "
> >   << pos.y << " " 
> >   << pos.x << " " 
> >   << std::endl;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And the global positions not to be the same. It just feels wrong.
> 
> But as explained, the position you're setting will relate to the origin. And the global bounds left & top position aren't the origin. Just like if you centre the origin on a square, then use `getGlobalBounds()` the left & top positions will not refer to the origin position of the shape.

Yes, but I also believe that if you don't set or change anything, the top & left position should relate to the top & left position.

**Comment by dogunbound** [comment]
After more debating, the real problem is documentation. The reason the offset is applied is that the letters in "Hello World!" are not the largest letters in the font. I didn't realize that. 

It would be largely helpful if the documentation could touch up on this.

**Comment by eXpl0it3r** [comment]
Fixed with #3568
